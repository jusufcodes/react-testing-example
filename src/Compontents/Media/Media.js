import React from "react";

export const Media = (props) => {
  return props.src ? <img src={props.src} alt={props.alt ? props.alt : ""} /> : null;
};
