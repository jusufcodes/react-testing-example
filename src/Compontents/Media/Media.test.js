import React from 'react';
import { shallow } from 'enzyme';
import renderer from "react-test-renderer";
import {Media} from "./Media";

describe("Card media component", () => {

  test("Card media without source", () => {
    const altText = "Placeholder Iamge";
    const avatar = <Media alt={altText} />;

    expect(shallow(avatar).isEmptyRender()).toEqual(true);

    const avatarSnapshot = renderer.create(avatar).toJSON();
    expect(avatarSnapshot).toMatchSnapshot();
  });

  test("Card media with source", () => {
    const image = "https://via.placeholder.com/300x200";
    const altText = "Placeholder Iamge";

    const media = <Media src={image} alt={altText} />;
    expect(shallow(media).prop("src")).toEqual(image);
    expect(shallow(media).prop("alt")).toEqual(altText);

    const mediaSnapshot = renderer.create(media).toJSON();
    expect(mediaSnapshot).toMatchSnapshot();
  });

});