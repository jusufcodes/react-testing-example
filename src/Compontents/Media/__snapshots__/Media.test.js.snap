// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card media component Card media with source 1`] = `
<img
  alt="Placeholder Iamge"
  src="https://via.placeholder.com/300x200"
/>
`;

exports[`Card media component Card media without source 1`] = `null`;
